content/python/concepts/enum/enum.md

```diff
@@ -0,0 +1,105 @@
+---
+Title: '`enum`'
+Description: 'Represents a set of unique, immutable constants in Python.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Data Types'
+  - 'Enum'
+  - 'Python'
+  - 'Variables'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+**`Enum`** (short for _enumeration_) is a data type in Python used to define a set of named, immutable constants.
+
+Enumerations make code more readable and maintainable by replacing **magic numbers** or strings with meaningful names.
+
+Enums are part of Python's built in `enum` module, introduced in Python 3.4.
+
+> **Note:** Magic numbers are unclear, hardcoded values in code. For example, `80` in a speed-checking program might be confusing. Replacing it with an enum constant, like `SpeedLimit.HIGHWAY`, makes the code easier to read and maintain.
+
+## Syntax
+
+```python
+from enum import Enum
+
+class EnumName(Enum):
+  MEMBER1 = value1
+  MEMBER2 = value2
+```
+
+- `EnumName`: The name of the enum class.
+- `MEMBER1`, `MEMBER2`: Names of the constants.
+- `value1`, `value2`: Values assigned to the constants (e.g. numbers or strings).
+
+## `enum` Module
+
+The `enum` module provides the `Enum` class for creating enums. It also includes:
+
+- `IntEnum`: Ensures values are integers.
+- `Flag`: Allows combining constants with bitwise operations.
+- `Auto`: Automatically assigns values.
+
+Enums provide methods like `.name` (the constant name) and `.value` (the assigned value).
```

Comment:
This can be written as:

Enums also provide methods like:
- `.name`: Returns the name of the enum member (as a string).
- `.value`: Returns the value assigned to the enum member.